aten/src/ATen/native/Normalization.cpp

```diff
@@ -83,6 +90,38 @@ Tensor batch_norm(
             running_mean, running_var, training, momentum, eps);
 }
 
+Tensor instance_norm(
+    const Tensor& input, const Tensor& weight /* optional */, const Tensor& bias /* optional */,
+    const Tensor& running_mean /* optional */, const Tensor& running_var /* optional */,
+    bool use_input_stats, double momentum, double eps, bool cudnn_enabled) {
+  if (!use_input_stats && (!running_mean.defined() || !running_var.defined())) {
+    throw std::runtime_error("Expected running_mean and running_var to be defined when use_input_stats is false");
+  }
+  std::vector<int64_t> shape = input.sizes().vec();
+  int64_t b = input.size(0);
+  int64_t c = input.size(1);
+  shape[1] = b * c;
+  shape[0] = 1;
+
+  Tensor weight_ = repeat_if_defined(weight, b);
+  Tensor bias_ = repeat_if_defined(bias, b);
+  Tensor running_mean_ = repeat_if_defined(running_mean, b);
+  Tensor running_var_ = repeat_if_defined(running_var, b);
+
+  auto input_reshaped = input.contiguous().view(shape);
```

Comment:
nit: you can probably replace this by `reshape`

Reply:
I think we should keep it view, the batch_norm op should actually reshape it if it's needed.